server/server.go

```diff
@@ -99,7 +100,11 @@ var (
 	etcdCommittedIndexGauge = etcdStateGauge.WithLabelValues("committedIndex")
 )
 
-// Server is the pd server.
+// if server doesn't implement all methods of basicsvr.Server, this line will result in
+// clear error message "*MyTServerype does not implement basicsvr.Server (missing Method method)"
+var _ basicsvr.Server = (*Server)(nil)
```

Comment:
Why to add it

Reply:
Would this check generate better error message during building?

Reply:
if Server didn't implement a method defined in basicsvr.Server, during compilation,  this line will result in the clear error message, something like:

`pkg/basic_server/basic_server.go:44:17: cannot use (*Server)(nil) (value of type *Server) as type TServer in variable declaration
pkg/basic_server/basic_server.go:79:79: undefined: basicsvr`

Without this line, it can pass the build, which seems bad to me.

Reply:
With this line, there is even error prompt at the line, if there is an method not implemented.

Reply:
I'm still new to golang and its conventional usage, but I'm wondering if it's better to have line as the best practice.

Reply:
@lhy1024 ^